### PR TITLE
[WIP] Excalidraw as an API / NPM package

### PR DESCRIPTION
**Pull request by ad1992**
- [ ]  Expose excalidraw as an API
- [ ] Create a umd build in npm packge
- [ ] create es/lib folders in npm package

Fixes https://github.com/excalidraw/excalidraw/issues/1196

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,21 +24,32 @@
       "last 1 safari version"
     ]
   },
+  "peerDependencies": {
+    "@sentry/browser": "^5.15.4",
+    "@sentry/integrations": "^5.15.4",
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
```
Comment (lipis) [inline package.json:31]: pin versions

Comment (ad1992) [reply package.json:31]: I haven't pinned these as its peer dependencies. If pinned the consumer library might end up installing multiple versions of these libraries. Does that make sense ?

**Comment by dwelle** [comment]
From our discussion over at https://github.com/excalidraw/excalidraw/pull/1493#pullrequestreview-403012598, I realized I haven't formed a good mental model of how this one would work (as I didn't really give this PR more than a cursory glance).

Now it makes me think if this is a good way to go because we'll not only need to keep in mind CSS namespacing, but pretty much everything.

For example, once there were talks about mocking `Math.random()` to hack around one issue --- no longer necessary, and that hack would only be done in a given event loop tick, but I mention this to illustrate a point that from now on we'd need to design Excalidraw in a way to ensure it plays well as an embedded system. I don't have any experience with this, but it feels like it could pose serious issues down the line.

/cc @vjeux @lipis

**Comment by ad1992** [comment]
closing this as these changes are made in [excalidraw-embed](https://github.com/excalidraw/excalidraw-embed)

**Comment by MichaelDeBoey** [comment]
@ad1992 I think `excalidraw-embed` isn't a public repo? 🤔

**Comment by dwelle** [comment]
@MichaelDeBoey not anymore, but it's not ready yet, so there's not much to see (it's basically a fork at this point).

**Pull request closed without merging**